plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/IncrementPropertyAction.java

```diff
@@ -16,56 +16,88 @@
  */
 package org.apache.unomi.plugins.baseplugin.actions;
 
+import java.lang.reflect.InvocationTargetException;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.Map;
 
+import org.apache.commons.beanutils.PropertyUtils;
 import org.apache.unomi.api.Event;
 import org.apache.unomi.api.Profile;
 import org.apache.unomi.api.actions.Action;
 import org.apache.unomi.api.actions.ActionExecutor;
 import org.apache.unomi.api.services.EventService;
 import org.apache.unomi.api.services.TopicService;
+import org.apache.unomi.persistence.spi.PropertyHelper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class IncrementInterestAction implements ActionExecutor {
+public class IncrementPropertyAction implements ActionExecutor {
 
-    private static final Logger LOG = LoggerFactory.getLogger(IncrementInterestAction.class.getName());
+    private static final Logger LOG = LoggerFactory.getLogger(IncrementPropertyAction.class.getName());
 
     private static final String EVENT_INTERESTS_PROPERTY = "interests";
-
     private static final String ACTION_INTERESTS_PROPERTY = "eventInterestProperty";
 
     private TopicService topicService;
-
     private EventService eventService;
-
     private Double interestsMinValue;
-
     private Double interestsMaxValue;
-
     private Double interestsDividerValue;
 
     @Override
     @SuppressWarnings("unchecked")
     public int execute(final Action action, final Event event) {
-        Map<String, Double> interestsAsMap = (Map<String, Double>) action.getParameterValues().get( ACTION_INTERESTS_PROPERTY );
+        final Profile profile = event.getProfile();
 
-        if ( interestsAsMap == null ) {
-            interestsAsMap = (Map<String, Double>) event.getProperty( EVENT_INTERESTS_PROPERTY );
+        Object propertyValue;
+        String propertyName = (String) action.getParameterValues().get("propertyName");
+        String parentPropertyName = propertyName.split("\\.")[0];
 
-            if (interestsAsMap == null) {
-                return EventService.NO_CHANGE;
+        try {
+            if (EVENT_INTERESTS_PROPERTY.equals(propertyName)) {
+                propertyValue = incrementInterests(action, event, profile);
```

Comment:
I think we should rename this method to processIncrementInterestEvent

Reply:
After reflection I wonder if we should split this action into 2 separate actions: IncrementInterests and IncrementProperty, because their logic is quite different.